modules/programs/hexchat.nix

```diff
@@ -0,0 +1,262 @@
+{ config, pkgs, lib, ... }:
+with builtins;
+with lib;
+let cfg = config.programs.hexchat;
+in {
+  meta.maintainers = [ maintainers.superherointj ];
+
+  options.programs.hexchat = {
+    channels = with lib.types;
+      let
+        modChannelOption = submodule {
+          options = {
+            autojoin = mkOption {
+              type = nullOr (listOf str);
+              description =
+                "Channels list to autojoin on connecting to server.";
+              example = ''[ "##linux" "#nix" ]'';
+              default = [ ];
+            };
+            charset = mkOption {
+              type = str;
+              description = "Charset";
+              default = "";
+              example = "UTF-8 (Unicode)";
+            };
+            commands = mkOption {
+              type = nullOr (listOf str);
+              description = "Commands to be executed on connecting to server.";
+              example = ''[ "ECHO Greetings fellow Nixer! ]'';
+              default = [ ];
+            };
+            loginMethod = mkOption {
+              type = with lib.types;
+                nullOr (types.enum [
+                  "nickServMsg"
+                  "nickServ"
+                  "challengeAuth"
+                  "sasl"
+                  "serverPassword"
+                  "saslExternal"
+                  "customCommands"
+                ]);
+              description = ''
+                null => Default
+                "nickServMsg" (1) => NickServ (/MSG NickServ + password)
+                "nickServ" (2) => NickServ (/NICKSERV + password)
+                "challengeAuth" (4) => Challenge Auth (username + password)
+                "sasl" (6) => SASL (username + password)
+                "serverPassword" (7) => Server password (/PASS password)
+                "saslExternal" (10) => SASL EXTERNAL (cert)
+                "customCommands" (9) => Custom... (connect commands)
```

Comment:
How does this custom works :thinking: ?

Reply:
In HexChat, Network Dialog, edit a specific network, in window there is a tab for "Connect commands" which is run when connected to channel. `customCommands` refers to that method. Example of command: "/msg NickServ IDENTIFY xxx". But could be anything really.

Reply:
I kinda imagined something like this, but maybe I didn't expressed myself well. How do I define a custom command in this module?

Reply:
It is defined in field "commands", declared in line 24:
```
            commands = mkOption {
              type = nullOr (listOf str);
              description = "Commands to be executed on connecting to server.";
              example = literalExample ''[ "ECHO Greetings fellow Nixer! ]'';
              default = [ ];
            };
```

Reply:
Do you have any further consideration about this? Can it be resolved?

Reply:
Ok, seems fine.

But I think it would be better to have this clarified in documentation.

Reply:
Expanded description to better describe it.

Reply:
LGTM.